test/core/annotations/test_annot_throws_exception.pkb

```diff
@@ -0,0 +1,187 @@
+            --%throws(-20145)
+            procedure raised_same_exception;
+
+            --%test(Throws one of the listed exceptions)
+            --%throws(-20145,-20146, -20189 ,-20563)
+            procedure raised_one_listed_exception;
+
+            --%test(Throws diff exception)
+            --%throws(-20144)
+            procedure raised_diff_exception;
+
+            --%test(Throws empty)
+            --%throws()
+            procedure empty_throws;
+
+            --%test(Ignores when only bad parameters are passed1)
+            --%throws(hello,784#,0-=234,,u1234)
+            procedure bad_paramters_with_except;
+
+            --%test(Ignores when only bad parameters are passed, the test does not raise a exception and it shows successful test)
+            --%throws(hello,784#,0-=234,,u1234)
+            procedure bad_paramters_without_except;
+
+            --%test(Detects a valid exception number within many invalid ones)
+            --%throws(7894562, operaqk, -=1, -1, pow74d, posdfk3)
+            procedure one_valid_exception_number;
+
+            --%test(Givess failure when a exception is expected and nothing is thrown)
+            --%throws(-20459, -20136, -20145)
+            procedure nothing_thrown;
+        end;
+    ';
+
+    l_package_body := '
+        create package body annotated_package_with_throws is
+            procedure raised_same_exception is
+            begin
+                raise_application_error(-20145, ''Test error'');
+            end;
+
+            procedure raised_one_listed_exception is
+            begin
+                raise_application_error(-20189, ''Test error'');
+            end;
+
+            procedure raised_diff_exception is
+            begin
+                raise_application_error(-20143, ''Test error'');
+            end;
+
+            procedure empty_throws is
+            begin
+                raise_application_error(-20143, ''Test error'');
+            end;
+
+            procedure bad_paramters_with_except is
+            begin
+                raise_application_error(-20143, ''Test error'');
+            end;
+
+            procedure bad_paramters_without_except is
+            begin
+                null;
+            end;
+
+            procedure one_valid_exception_number is
+            begin
+              raise dup_val_on_index;
+            end;
+
+            procedure nothing_thrown is
+            begin
+                null;
+            end;
+        end;
+    ';
+
+    execute immediate l_package_spec;
+    execute immediate l_package_body;
+  end;
+
+  procedure drop_package is
+    pragma autonomous_transaction;
+
+    l_drop_statment VARCHAR2(32737);
+  begin
+    l_drop_statment := 'drop package annotated_package_with_throws';
+    execute immediate l_drop_statment;
+  end;
+
+  function execution_test_result(a_procedure_test_name in varchar) return varchar2 is
+    l_result varchar2(32737);
+  begin
+    select column_value
```

Comment:
Maybe run the suite once in the `beforeall`?
That way the tests will run much much faster.
Currently each individual test takes ~0.5 second to execute and the first one needs over 2 seconds (includes suite parsing for annotated_package_with_throws)
```
      annotations- throws
        Gives success when annotated number exception is thrown [2.606 sec]
        Gives succes when one of the annotated number exceptions is thrown [.474 sec]
        Gives failure when the raised exception is different that the annotated one [.457 sec]
        Ignores when  the annotation throws is empty [.488 sec]
        Ignores when only bad parameters are passed, the test raise a exception and it shows errored test [.524 sec]
        Ignores when only bad parameters are passed, the test does not raise a exception and it shows successful test [.491 sec]
        Detects a valid exception number within many invalid ones [.484 sec]
        Givess failure when a exception is expected and nothing is thrown [.529 sec]
```